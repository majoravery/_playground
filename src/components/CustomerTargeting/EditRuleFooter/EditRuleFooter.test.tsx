// import React from 'react';
// import { render, fireEvent } from '@testing-library/react';
// import { default as EditRuleFooter, CustomerTargetingEditFooterProps } from './EditRuleFooter';

// const defaultComponentProps: CustomerTargetingEditFooterProps = {
//   onApply: jest.fn(),
//   onCancel: jest.fn(),
// };

// describe('EditRuleFooter', () => {
//   it('renders', () => {
//     const { asFragment } = render(<EditRuleFooter {...defaultComponentProps} />);
//     expect(asFragment()).toMatchSnapshot();
//   });

//   it('renders with apply button disabled', () => {
//     const { asFragment } = render(<EditRuleFooter {...defaultComponentProps} disableApplyButton />);
//     expect(asFragment()).toMatchSnapshot();
//   });

//   it('renders with remove button', () => {
//     const { asFragment } = render(<EditRuleFooter {...defaultComponentProps} deleteButtonText='Remove' />);
//     expect(asFragment()).toMatchSnapshot();
//   });

//   it('triggers onApply click', () => {
//     const onApplyMock = jest.fn();
//     const { getByTestId } = render(<EditRuleFooter {...defaultComponentProps} onApply={onApplyMock} />);
//     const selectButton = getByTestId('tab-content-footer-select-btn');
//     fireEvent.click(selectButton);
//     expect(onApplyMock).toHaveBeenCalled();
//   });

//   it('triggers onCancel click', () => {
//     const onCancelMock = jest.fn();
//     const { getByTestId } = render(<EditRuleFooter {...defaultComponentProps} onCancel={onCancelMock} />);
//     const removeButton = getByTestId('tab-content-footer-remove-btn');
//     fireEvent.click(removeButton);
//     expect(onCancelMock).toHaveBeenCalled();
//   });
// });
export {};
