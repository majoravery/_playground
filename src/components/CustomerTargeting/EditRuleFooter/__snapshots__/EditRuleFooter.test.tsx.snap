// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRuleFooter renders 1`] = `
<DocumentFragment>
  <div
    class="tab-content-footer"
  >
    <button
      class="button--component tab-content-footer-remove-btn btn btn-outline-primary"
      data-testid="tab-content-footer-remove-btn"
      type="button"
    >
      Cancel
    </button>
    <button
      class="button--component tab-content-footer-select-btn btn btn-primary"
      data-testid="tab-content-footer-select-btn"
      type="button"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;

exports[`EditRuleFooter renders with apply button disabled 1`] = `
<DocumentFragment>
  <div
    class="tab-content-footer"
  >
    <button
      class="button--component tab-content-footer-remove-btn btn btn-outline-primary"
      data-testid="tab-content-footer-remove-btn"
      type="button"
    >
      Cancel
    </button>
    <button
      class="button--component tab-content-footer-select-btn btn btn-primary"
      data-testid="tab-content-footer-select-btn"
      disabled=""
      type="button"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;

exports[`EditRuleFooter renders with remove button 1`] = `
<DocumentFragment>
  <div
    class="tab-content-footer"
  >
    <button
      class="button--component tab-content-footer-remove-btn btn btn-outline-danger"
      data-testid="tab-content-footer-remove-btn"
      type="button"
    >
      <svg
        fill="none"
        height="14"
        opacity="1"
        stroke="#051022"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
      <span
        class="text-with-icons"
      >
        Remove
      </span>
    </button>
    <button
      class="button--component tab-content-footer-select-btn btn btn-primary"
      data-testid="tab-content-footer-select-btn"
      type="button"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
